File: Server/Models/UsedDates.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
namespace Server.Models
{
    public class UsedDates
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [Required]
        public string EquipmentId { get; set; }
        public Equipment? Equipment { get; set; }

        [Required]
        public string UserId { get; set; }
        public ApplicationUser? User { get; set; }

        [Required]
        public DateTime StartDate { get; set; }

        [Required]
        public DateTime EndDate { get; set; }

        [Required]
        public string Type { get; set; } = "Taken"; // Default to "Taken"

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }
    }
} 

File: Client/src/types/UsedDates.ts
export type UsedDateType = 'Taken' | 'Planning' | 'Wish';

export interface UsedDates {
    id?: number;
    equipmentId: string;
    userId: string;
    startDate: Date;
    endDate: Date;
    type: UsedDateType;
    createdAt: Date;
    updatedAt: Date;
}

export interface CreateUsedDatesDto {
    equipmentId: string;
    startDate: Date;
    endDate: Date;
    type: UsedDateType;
}

export interface UpdateUsedDatesDto {
    startDate?: Date;
    endDate?: Date;
    type?: UsedDateType;
} 